src/components/model/ModelSpecification.svelte

```diff
@@ -0,0 +1,268 @@
+        cellDataType: 'boolean',
+        editable: hasEditSpecPermission,
+        headerName: '',
+        resizable: false,
+        suppressAutoSize: true,
+        suppressSizeToFit: true,
+        valueGetter: (params: ValueGetterParams<BaseMetadata>) => {
+          const { data } = params;
+          if (data) {
+            return !!selectedSpecifications[data.id]?.selected;
+          }
+          return false;
+        },
+        width: 35,
+      },
+    ];
+  }
+  $: if (selectedSpecifications) {
+    dataGrid?.redrawRows();
+  }
+
+  function onNewMetadata() {
+    dispatch('newMetadata');
+  }
+
+  function onSelectDefinition(event: CustomEvent<BaseMetadata[]>) {
+    const { detail: selectedMetadata } = event;
+    if (selectedMetadata.length > 0) {
+      dispatch('selectSpecification', {
+        id: selectedMetadata[0].id,
+        revision: selectedMetadata[0].versions[0].revision,
+      });
+    } else {
+      dispatch('selectSpecification', null);
+    }
+  }
+
+  function onViewMetadata({ id }: Pick<BaseMetadata, 'id'>) {
+    dispatch('viewMetadata', id);
+  }
+
+  function onToggleSpecification(event: CustomEvent<CellEditingStoppedEvent<BaseMetadata, boolean>>) {
+    const {
+      detail: { data, newValue },
+    } = event;
+
+    if (data) {
+      dispatch('toggleSpecification', {
+        id: data.id,
+        selected: !!newValue,
+      });
+    }
+  }
+</script>
+
+<div class="metadata-container">
+  <div class="metadata-filter-container">
+    <div class="metadata-title">{formattedType}s</div>
+    <Input>
+      <input bind:value={filterText} class="st-input" placeholder={`Filter ${metadataType}s`} style="width: 100%;" />
+    </Input>
+    <button
+      class="st-button secondary ellipsis"
+      use:permissionHandler={{
+        hasPermission: hasCreatePermission,
+        permissionError,
+      }}
+      on:click={onNewMetadata}
+    >
+      New
+    </button>
+  </div>
+  <hr />
+  <div class="metadata-table-container">
+    {#if filteredMetadata.length}
+      <DataGrid
+        bind:this={dataGrid}
+        {columnDefs}
+        rowData={filteredMetadata}
+        rowSelection="single"
+        selectedRowIds={selectedSpecification ? [selectedSpecification.id] : []}
+        on:cellEditingStopped={onToggleSpecification}
+        on:selectionChanged={onSelectDefinition}
+      />
+    {:else}
+      <div class="p1 st-typography-label">No {formattedType}s Found</div>
+    {/if}
+  </div>
+</div>
+
+<style>
+  .metadata-container {
+    display: grid;
+    grid-template-rows: min-content min-content auto;
+    row-gap: 0.5rem;
+  }
+
+  .metadata-container hr {
+    border: none;
+    border-top: 1px solid #e0e0e0;
+    margin: 0 1rem;
+    width: auto;
+  }
+
+  .metadata-filter-container {
+    align-items: center;
+    column-gap: 0.25rem;
+    display: grid;
+    grid-template-columns: min-content auto min-content;
+    margin: 0.5rem 1rem 0;
+  }
+
+  .metadata-title {
+    font-weight: bold;
```

Comment:
Could instead use the `st-typography-bold` class?

Reply:
Good call! It's hard to keep track of what styles we have predefined in stellar...